main.py

```diff
@@ -1,318 +1,347 @@
+        return bool(re.match(pattern, email))
+
+    accountPath = Utils.getProjectRoot() / "accounts.json"
+    if not accountPath.exists():
+        accountPath.write_text(
+            json.dumps(
+                [{"username": "Your Email", "password": "Your Password"}], indent=4
+            ),
+            encoding="utf-8",
+        )
+        noAccountsNotice = """
+    [ACCOUNT] Accounts credential file "accounts.json" not found.
+    [ACCOUNT] A new file has been created, please edit with your credentials and save.
+    """
+        logging.warning(noAccountsNotice)
+        exit(1)
+    loadedAccounts: list[Account] = []
+    for rawAccount in json.loads(accountPath.read_text(encoding="utf-8")):
+        account: Account = Account(**rawAccount)
+        if not validEmail(account.username):
+            logging.warning(
+                f"[CREDENTIALS] Invalid email: {account.username}, skipping this account"
+            )
+            continue
+        loadedAccounts.append(account)
+    random.shuffle(loadedAccounts)
+    return loadedAccounts
+
+
+class AppriseSummary(Enum):
+    always = auto()
+    on_error = auto()
+    never = auto()
+
+
+def executeBot(currentAccount: Account, args: argparse.Namespace):
+    logging.info(f"********************{currentAccount.username}********************")
+
+    startingPoints: int
+    accountPoints: int
+    remainingSearches: RemainingSearches
+    goalTitle: str
+    goalPoints: int
+
+    if args.searchtype in ("desktop", None):
+        with Browser(mobile=False, account=currentAccount, args=args) as desktopBrowser:
+            utils = desktopBrowser.utils
+            Login(desktopBrowser, args).login()
+            startingPoints = utils.getAccountPoints()
+            logging.info(
+                f"[POINTS] You have {utils.formatNumber(startingPoints)} points on your account"
+            )
+            # todo Send notification if these fail to Apprise versus just logging
+            ReadToEarn(desktopBrowser).completeReadToEarn(startingPoints)
+            DailySet(desktopBrowser).completeDailySet()
+            PunchCards(desktopBrowser).completePunchCards()
+            MorePromotions(desktopBrowser).completeMorePromotions()
+            # VersusGame(desktopBrowser).completeVersusGame()
+
+            with Searches(desktopBrowser) as searches:
+                searches.bingSearches()
+
+            goalPoints = utils.getGoalPoints()
+            goalTitle = utils.getGoalTitle()
+
+            remainingSearches = desktopBrowser.getRemainingSearches(
+                desktopAndMobile=True
+            )
+            accountPoints = utils.getAccountPoints()
+
+    if args.searchtype in ("mobile", None):
+        with Browser(mobile=True, account=currentAccount, args=args) as mobileBrowser:
+            utils = mobileBrowser.utils
+            Login(mobileBrowser, args).login()
+            if startingPoints is not None:
```

Comment:
This causes an error when only searching with mobile only since `startingPoints` does not get initialized. `startingPoints: int` does not create the variable so when trying to access it will result in `NameError` being thrown. A simple fix for this would just be gating its assignment based on whether `searchtype` is mobile or None.

Reply:
Good catch, was my intent with the if condition but need to negate and initialize startingPoints to None. Should be fixed now.

Reply:
I plan to update the ReadToEarn section and change it to MobileApp. It should be in the mobile section or a section of its own. 

I want get the Bing mobile app-only check-in working as a good amount of extra points. Might get other points via API calls instead of selenium, but not sure it will work.